File: perl/bin/export.pl
use strict;
use Cwd qw(abs_path);
use File::Basename;

my %options = ();

BEGIN{
	my $ruta = abs_path(join('/', dirname(__FILE__), '..', 'lib'));

	push @INC, $ruta;	
}


use Validator::Exporter;

foreach my $arg (@ARGV){
    my ($key, $value) = split /\=/, $arg;
    $options{$key} = $value;
}

unless(@ARGV && $options{package_name}){

    print "Usage: $0 package_name=<name> validations_file=<path to validations.yml>\n";
    exit 1;
}



print Validator::Exporter->new(%options)->dumpPackage;

File: ruby/bin/export.rb
$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), '..', 'lib'))

require 'validator/exporter'

options = {}

ARGV.each do |arg|

  key, value = arg.split('=', 2)

  options[key.to_sym] = value

end

unless ARGV && options.has_key?(:package_name)

    print "Usage: #{$0} package_name=<name> validations_file=<path to validations.yml>\n";
    exit 1;
end

print Validator::Exporter.new(options).dumpPackage
